### PR TITLE
fix(proxy_restart): Consistent way to restart nodes with proxy

### LINKED ISSUE #3134: proxy_restart.py is flaky, fails with "Connect call failed ('127.0.0.1', 24578)"
E.g

http://52.149.162.182:3000/#/test/6792
http://52.149.162.182:3000/#/test/7512

### PR DESCRIPTION
**Pull request by mfornet**
On proxy restart wait for the previous instance to get closed before starting a new one

fixes #3134 

Test plan
=======
sanity/proxy_restart.py pass consistently

http://nayduck.eastus.cloudapp.azure.com:3000/#/run/285

### CODE REVIEW

File: pytest/lib/proxy.py [hunk 1]
```diff
@@ -326,6 +327,10 @@ def proxify_node(node, ps, handler, global_stopped, error, proxy):
     outer_port = inner_port + 100
 
     def start_proxy():
+        # local_stopped denotes the current of state of the proxy:
+        # 0: The proxy is running
+        # 1: The proxy is running but should be closed soon
```
Comment (bowenwang1996) [inline pytest/lib/proxy.py:332]: where is 1 used

Comment (mfornet) [reply pytest/lib/proxy.py:332]: To kill a node, this value is set to `1` so the underlying proxy finishes

https://github.com/nearprotocol/nearcore/pull/3272/files#diff-6fc0b6294fe72a99a922a578c7bf0561L343

**Review by SkidanovAlex — APPROVED** [review]

File: pytest/lib/proxy.py [hunk 2]
```diff
@@ -44,7 +44,7 @@
 MSG_TIMEOUT = 10
 _MY_PORT = [None]
 
-logging.basicConfig(format='%(asctime)s %(message)s', level=logging.INFO)
+logging.basicConfig(format='%(asctime)s %(message)s', level=logging.DEBUG)
```
Comment (SkidanovAlex) [inline pytest/lib/proxy.py:47]: Do you want to keep it at `DEBUG`?

**Review by bowenwang1996 — APPROVED** [review]

File: pytest/lib/proxy.py (hunk 2 above)
Comment (mfornet) [reply pytest/lib/proxy.py:47]: No

**Pull request merged**